### PR TITLE
Add code for A000688 / abelian groups

### LINKED ISSUE #94: A000688 / abelian-groups
[A000688 / Number of Abelian groups of order n; number of factorizations of n into prime powers.](https://oeis.org/A000688)

The sequence is not implemented yet.

<code>
1 1 1 2 1 1 1 3 2 1 1 2 1 1 1 5 1 2 1 2 1 1 1 3 2 1 3 2 1 1 1 7 1 1 1 4 1 1 1 3 1 1 1 2 2 1 1 5 2 2 1 2 1 3 1 3 1 1 1 2 1 1 2 11 1 1 1 2 1 1 1 6 1 1 2 2 1 1 1 5 5 1 1 2 1 1 1 3 1 2 1 2 1 1 1 7 1 2 2 4 1 1 1 3 1 1 1
</code>

### PR DESCRIPTION
**Pull request by thundergnat**
I'm of two minds about this. It nominally relies on A000041 which is
not yet implemented. However I've put in a hard coded list which will
allow up to 2**173526 terms before it fails. Yeah, it's not infinite,
but it's probably good enough for all but the most extreme use.

Resolves #94

### CODE REVIEW

File: lib/Math/Sequences/Integer.pm6 [hunk 1]
```diff
@@ -509,7 +509,14 @@ our @A000609 is export = 1, &NOSEQ ... *;
 # A000670 / preferential arrangements
 our @A000670 is export = 1, &NOSEQ ... *;
 # A000688 / abelian groups
-our @A000688 is export = 1, &NOSEQ ... *;
+our @A000688 is export our @A000688 = (1..*).map: {
```
Comment (AlexDaniel) [inline lib/Math/Sequences/Integer.pm6]: There is some issue here.

File: lib/Math/Sequences/Integer.pm6 [hunk 2]
```diff
@@ -509,7 +509,14 @@ our @A000609 is export = 1, &NOSEQ ... *;
...
+    # @A000041 NYA. Hardcoded list, will fail at term 2 ** 173526
```
Comment (AlexDaniel) [inline lib/Math/Sequences/Integer.pm6]: NYA?

nya

**Comment by AlexDaniel** [comment]
> However I've put in a hard coded list which will allow up to 2**173526 terms before it fails. Yeah, it's not infinite

That's long enough. It's totally fine.

**Comment by thundergnat** [comment]
Sigh. I think I may need to stop for a while, I'm starting to make (even more) stupid errors.

File: lib/Math/Sequences/Integer.pm6 (hunk 2 above)
Comment (thundergnat) [reply lib/Math/Sequences/Integer.pm6]: I was very confused for a short while until I figured out what you were referring to. :-)

File: lib/Math/Sequences/Integer.pm6 (hunk 1 above)
Comment (thundergnat) [reply lib/Math/Sequences/Integer.pm6]: fix pushed

File: lib/Math/Sequences/Integer.pm6 [hunk 3]
```diff
@@ -509,7 +519,14 @@ our @A000609 is export = 1, &NOSEQ ... *;
 # A000670 / preferential arrangements
 our @A000670 is export = 1, &NOSEQ ... *;
 # A000688 / abelian groups
-our @A000688 is export = 1, &NOSEQ ... *;
+our @A000688 = (1..*).map: {
```
Comment (AlexDaniel) [inline lib/Math/Sequences/Integer.pm6]: `is export` is missing here, but I'll add it myself.

**Comment by AlexDaniel** [comment]
Thanks!

**Comment by AlexDaniel** [comment]
Actually, `A000688` is wrong.

```
expected: '1 1 1 2 1 1 1 3 2 1 1 2 1 1 1 5 1 2 1 2 1 1 1 3 2 1 3 2 1 1 1 7 1 1 1 4 1 1 1 3 1 1 1 2 2 1 1 5 2 2 1 2 1 3 1 3 1 1 1 2 1 1 2 11 1 1 1 2 1 1 1 6 1 1 2 2 1 1 1 5 5 1 1 2 1 1 1 3 1 2 1 2 1 1 1 7 1 2 2 4 1 1 1 3 1 1 1'
     got: '1 1 1 2 1 1 1 3 2 1 1 2 1 1 1 5 1 2 1 2 1 1 1 3 2 1 3 2 1 1 1 7 1 1 1 2 1 1 1 3 1 1 1 2 2 1 1 5 2 2 1 2 1 3 1 3 1 1 1 2 1 1 2 11 1 1 1 2 1 1 1 3 1 1 2 2 1 1 1 5 5 1 1 2 1 1 1 3 1 2 1 2 1 1 1 7 1 2 2 2 1 1 1 3 1 1 1'
```

**Pull request merged**